src/tests/mam4_coagulation_unit_tests.cpp

```diff
@@ -21,10 +21,162 @@ TEST_CASE("test_constructor", "mam4_coagulation_process") {
   REQUIRE(process.aero_config() == mam4_config);
 }
 
+TEST_CASE("bm0ij_data", "mam4_cagulation_process") {
+
+  // Here we test a few values returned directly from fortran with those
+  // as defined in the C++ code.
+
+  int n1 = 0;
+  int n2a = 1;
+  int n2n = 2;
+  Real bm0ij_f = 0.674432;
+  Real bm0ij_c = coagulation::bm0ij_data(n1, n2a, n2n);
+  REQUIRE(bm0ij_f == bm0ij_c);
+
+  n1 = 1;
+  n2a = 4;
+  n2n = 0;
+  bm0ij_f = 0.739575;
+  bm0ij_c = coagulation::bm0ij_data(n1, n2a, n2n);
+  REQUIRE(bm0ij_f == bm0ij_c);
+}
+
+TEST_CASE("intermodal_coag_rate_for_0th_moment", "mam4_coagulation_process") {}
+
+TEST_CASE("bm3ij_data", "mam4_coagulation_process") {
+
+  int n1 = 0;
+  int n2a = 1;
+  int n2n = 2;
+  Real bm3i_f = 0.74927;
+  Real bm3i_c = coagulation::bm3i_data(n1, n2a, n2n);
+  REQUIRE(bm3i_f == bm3i_c);
+
+  n1 = 1;
+  n2a = 4;
+  n2n = 0;
+  bm3i_f = 0.91886;
+  bm3i_c = coagulation::bm3i_data(n1, n2a, n2n);
+  REQUIRE(bm3i_f == bm3i_c);
+}
+
+TEST_CASE("intermodal_coag_rate_for_3rd_moment", "mam4_coagulation_process") {}
+
+TEST_CASE("intra_coag_rate_for_0th_moment", "mam4_coagulation_process") {
+
+  Real a_const = 0.0;
+  Real knc = 0.0;
+
+  Real kngxx = 0.0;
+  Real kfmxx = 0.0;
+  Real sqdgxx = 0.0;
+
+  Real esxx04 = 0.0;
+  Real esxx08 = 0.0;
+  Real esxx20 = 0.0;
+
+  Real esxx01 = 0.0;
+  Real esxx05 = 0.0;
+  Real esxx25 = 0.0;
+
+  int n2x = 1;
+  Real qnxx = 0.0;
+
+  coagulation::intramodal_coag_rate_for_0th_moment(
+      a_const, knc, kngxx, kfmxx, sqdgxx, esxx04, esxx08, esxx20, esxx01,
+      esxx05, esxx25, n2x, qnxx);
+}
+
+TEST_CASE("get_coags", "mam4_coagulation_process") {}
+
+TEST_CASE("getcoags_wrapper_f", "mam4_coagulation_process") {}
+
+TEST_CASE("mam_coag_aer_update", "mam4_coagulation_process") {}
+
+TEST_CASE("update_qnum_for_intra_and_intermodal_coag",
+          "mam4_coagulation_process") {}
+
+TEST_CASE("update_qnum_for_intramodal_coag", "mam_coagulation_process") {}
+
+TEST_CASE("mam_coag_num_update", "mam_coagulation_process") {}
+
+TEST_CASE("mam_coag_1subarea", "mam_coagulation_process") {}
+
 TEST_CASE("test_compute_tendencies", "mam4_coagulation_process") {
 
-  // Kokkos::parallel_for(
-  //     team_policy, KOKKOS_LAMBDA(const ThreadTeam &team) {
-  //       process.compute_tendencies(team, t, dt, atm, progs, diags, tends);
-  //     });
+  ekat::Comm comm;
+  ekat::logger::Logger<> logger("aging unit tests",
+                                ekat::logger::LogLevel::debug, comm);
+  int nlev = 72;
+  Real pblh = 1000;
+  Atmosphere atm(nlev, pblh);
+  mam4::Prognostics progs(nlev);
+  mam4::Diagnostics diags(nlev);
+  mam4::Tendencies tends(nlev);
+
+  mam4::AeroConfig mam4_config;
+  mam4::NucleationProcess process(mam4_config);
+
+  const auto prog_qgas0 = progs.q_gas[0];
+  const auto tend_qgas0 = tends.q_gas[0];
+  auto h_prog_qgas0 = Kokkos::create_mirror_view(prog_qgas0);
+  auto h_tend_qgas0 = Kokkos::create_mirror_view(tend_qgas0);
+  Kokkos::deep_copy(h_prog_qgas0, prog_qgas0);
```

Comment:
Check this deep_copy: 
data from device is copied to host, but I believe you need to copy from host to device. Also, values in device are initialize to zeros.

Reply:
@jeff-cohere Do you want to take a look at this, I essentially copied this bit of code from mam4_nucleation_unit_tests.cpp.

Reply:
Maybe it would be more instructive to copy test code from `calcsize` or `gasaerexch` or another process that uses several aerosol species. The nucleation parameterization was the first one we attacked and it only really adds sulfate to the aitken mode. And the testing on the other parameterizations is definitely more thorough at this point--I'm going to have to revisit nucleation and raise it to the standard set by others!

Reply:
In consultation with @jeff-cohere  I'm going to opt for improving the testing beyond a GPU smoke tests in another PR.